plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -547,13 +552,15 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             }
 
             if (except_ptr) {
+               std::string details(except_ptr->to_detail_string(), 0, _max_size_of_detail_exception_msg);
```

Comment:
`fc::format_string` supports a `bool minimize` which could be used instead. Then there is no need for another max size of string variable.

The log output should not include newlines. I think `top_message()` provides what you need, if not you may need to modify `fc::exception`.
There is a comment https://github.com/eosnetworkfoundation/mandel/issues/683#issuecomment-1186546703 asking for contract name and action name to be included. When a trace is available seems like you could just print that info from the last action trace in order to provide it.

Reply:
Thanks for the comments! I will incorporate them.